### PR TITLE
feat: add support for Auto IAM AuthN

### PR DESCRIPTION
**Pull request by enocom**

### CODE REVIEW

**Comment by adamstrawson** [comment]
What's the priority for getting this rolled out, noticed there hasn't been any progress in the last month. 

We're starting to create a plan on shifting all our workloads from CloudSQL to AlloyDB, and would be nice to be able to retain IAM in the same way we do with CloudSQL Proxy at the moment, so that we don't have to adopt user/pass auth to bridge that gap..

**Comment by enocom** [comment]
@adamstrawson this feature is being rolled out on the backend, but has hit some delays. We're hoping to ship this ASAP, but don't have a ETA presently. Let me check with the team to see how things are going.

File: cmd/root.go [hunk 1]
```diff
@@ -184,8 +203,29 @@ Instance Level Configuration
     instances, the proxy will ensure that the last path element is
     '.s.PGSQL.5432' appending it if necessary. For example,
 
-        ./cloud-sql-proxy \
-          'my-project:us-central1:my-db-server?unix-socket-path=/path/to/socket'
+        ./alloydb-aith-proxy \
```
Comment (jackwotherspoon) [inline cmd/root.go:206]: nit:
```suggestion
        ./alloydb-auth-proxy \
```

**Review by jackwotherspoon — APPROVED** [review]
LGTM 🚀

File: internal/proxy/proxy.go [hunk 2]
```diff
@@ -64,6 +68,11 @@ type Config struct {
 	// API.
 	UserAgent string
 
+	// AutoIAMAuthN enabled automatic IAM authentication which results in the
```
Comment (jackwotherspoon) [inline internal/proxy/proxy.go:71]: nit: enables?
```suggestion
	// AutoIAMAuthN enables automatic IAM authentication which results in the
```

File: cmd/root.go (hunk 1 above)
Comment (nancynh) [reply cmd/root.go:206]: It still looks the same?

**Review by nancynh — APPROVED** [review]
LGTM, thanks Eno!

**Pull request merged**